### DISCUSSION TITLE
How to automate wazuh deployment

### DISCUSSION THREAD

**Discussion by Odiesinfosec** [Security-Onion-Solutions/securityonion#9823] [Unsupported Versions] [answered]
As title says, I am struggling to configure wazuh on so2 to be able to deployed with a GPO. I can not connect on our 500+ hosts.
I do not have time or patience to manually run the MSI, add an agent with so-wazuh-agent-manage, extract the key, go back to the host and import the key, there for sure is a better way to do this but the documentation leads to more questions than answers. 
There is documentation about using ossec-authd... But how does that work? It looks like a command yet when I try to use it I get "command not found" again, the doc has no other information about this or how to run this command.
my two questions are,
How can I configure wazuh to use password authentication and, How do I let the agents register themselves instead of running so-wazuh-agent-manage and manually adding an agent for every single host?
When you deploy wazuh, how did you do it in a large network without doing it 1 by 1?
Thanks

**Comment by InfosecGoon** [comment, marked as answer]
Assuming you're connecting your Windows hosts to a standalone or manager node, ossec-authd should already been running and accessible on it. You'd need to push the MSI to the endpoint using a GPO or your package management solution of choice, and then run something like this:

.\wazuh-agent-3.13.1-1.msi /q WAZUH_MANAGER="192.168.182.202" WAZUH_REGISTRATION_SERVER="192.168.182.202" WAZUH_AGENT_NAME="W2012R2"

**Reply by InfosecGoon** [reply]
I should probably also point out that Wazuh has a limited lifespan left -- version 2.4 will be replacing it with Elastic Agent, so this might not be a good time for a large deployment if it can wait a few months.

**Reply by Odiesinfosec** [reply]
Thank you InfosecGoon! For now I will just push out osquery agents and elasticsearch/winlogbeat!
cheers

**Reply by morohyahaya** [reply]
hello please I have a question concerning this subject based on what you gave as an answer how will the addition of the keys be done I tried the case of a single machine when I installed the executable (msi) I have after an interface on which I put the address of the server and a key and the key I got it when I added the ip address of the agent in this parameter so-wazuh- agent-manage of the server then I manage to have the key now my question is how did it go for this part I must add all the agents then recover their key and add them to the authentication interface or else
Thanks in advance .
